utils/errors.ts

```diff
@@ -0,0 +1,113 @@
+ *  - is not valid JSON / PDF
+ *  - when JSON, is not of the expected type (for example, doesn't look like PortfolioStep/FundingStep/ApplicationStep)
+ */
+export const REQUEST_BODY_INVALID = new OtherErrorResponse(
+  "A valid request body must be provided",
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when a request parameter exists but is invalid
+ * Could be invalid because:
+ *  - has the wrong form (for example, uuid or date expected but won't parse)
+ *  - is out of range (for example, integer not in the accepted range)
+ */
+export const QUERY_PARAM_INVALID = new OtherErrorResponse("Invalid request parameter", ErrorStatusCode.BAD_REQUEST);
+
+/**
+ * To be used when a required path parameter is not received (not present or empty)
+ */
+export const PATH_PARAMETER_REQUIRED_BUT_MISSING = new OtherErrorResponse(
+  "Required path parameter is missing",
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when the specified Portfolio Draft is not found
+ */
+export const NO_SUCH_PORTFOLIO_DRAFT_404 = new OtherErrorResponse(
+  "Portfolio Draft with the given ID does not exist",
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when a specified Portfolio Draft does not exist
+ */
+export const NO_SUCH_PORTFOLIO_DRAFT_400 = new OtherErrorResponse(
+  "The given Portfolio Draft does not exist",
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when a Portfolio Step is not found for a specified Portfolio Draft
+ */
+export const NO_SUCH_PORTFOLIO_STEP = new OtherErrorResponse(
+  "Portfolio Step not found for this Portfolio Draft",
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when a Funding Step is not found for a specified Portfolio Draft
+ */
+export const NO_SUCH_FUNDING_STEP = new OtherErrorResponse(
+  "Funding Step not found for this Portfolio Draft",
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when a Application Step is not found for a specified Portfolio Draft
+ */
+export const NO_SUCH_APPLICATION_STEP = new OtherErrorResponse(
+  "Application Step not found for this Portfolio Draft",
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when a Portfolio or Application is not found when creating a new environment
+ */
+export const NO_SUCH_PORTFOLIO_OR_APPLICATION = new OtherErrorResponse(
+  "Portfolio or Application with the given ID does not exist",
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when an environment in a specific application already has the same name
+ */
+export const DUPLICATE_ENVIRONMENT_NAME = new OtherErrorResponse(
+  "Environment name already exists in this application",
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when a function has not been implemented
+ */
+export const NOT_IMPLEMENTED = new OtherErrorResponse("Not implemented", ErrorStatusCode.NOT_FOUND);
+
+/**
+ * To be used when a function has not been implemented
+ */
+export const PORTFOLIO_ALREADY_SUBMITTED = new OtherErrorResponse(
```

Comment:
We probably don't want to add a bunch of errors that we aren't using yet. Thoughts?

Reply:
Yeah, I agree. There were other derivative errors I removed too, so I think this should be done for other errors not being used.

Reply:
Reduced to 2 errors and removed a decent amount in `error-handling-middleware`.